### PR TITLE
[2.18.x] DDF-5439 Decouple UI and Service Facet Whitelist

### PR DESCRIPTION
**Pull request by willwill96**
#### What does this PR do?
Backport of https://github.com/codice/ddf/pull/5440
#### Who is reviewing it? 
@andrewzimmer 
@hayleynorton 
@zta6 

#### Ask 2 committers to review/merge the PR and tag them here.
@djblue 
@bdeining 

#### How should this be tested?
See master PR
#### What are the relevant tickets?
Fixes: #5439 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by lamhuy** [comment]
build now

**Comment by willwill96** [comment]
build now

**Comment by willwill96** [comment]
build now

**Comment by hayleynorton** [comment]
Hero successful: same changes as verified in 2.19.x

File: catalog/plugin/catalog-plugin-facet-attribute-access/pom.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    <packaging>bundle</packaging>
+
+    <dependencies>
+        <dependency>
+            <groupId>ddf.catalog.core</groupId>
+            <artifactId>catalog-core-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>ddf.catalog.core</groupId>
+            <artifactId>catalog-core-api-impl</artifactId>
```
Comment (bdeining) [inline catalog/plugin/catalog-plugin-facet-attribute-access/pom.xml]: i think this needs to be embedded

**Pull request merged**